docs/trusted-publishers-for-all-package-repositories.md

```diff
@@ -0,0 +1,122 @@
+* Configuring a Trusted Publisher is a means of establishing trust in some external state. That external state (both the IdP itself and the resources managed by the IdP) must not be controllable by untrusted parties.
+
+It’s recommended to document this security model alongside provider-specific guides on how to adopt Trusted Publishers for the package repository.
+
+### Data model of Trusted Publishers
+
+The intuitive answer for the data model of Trusted Publisher configurations to packages is one-to-one, with one source of artifacts for each package. However, for many packaging ecosystems, a many-to-many approach will be needed to support the complete range of existing use-cases for authenticating a source of artifacts with a project.
+
+“Monorepos” are where multiple separate projects’ source code is stored in a single source repository. Monorepos are common across many packaging ecosystems and their existence requires a one-to-many relationship between a Trusted Publisher configuration and packages on a package repository.
+
+For packaging ecosystems where there are multiple artifacts per release or multiple publishing workflows to support multiple platforms or CPU architectures (such as Python), the opposite “one-to-many” relationship is needed to support multiple Trusted Publisher configurations for a single package, to allow publishing from multiple build systems
+
+### Identity Providers and Claims
+
+OIDC tokens are implemented as [JSON Web Tokens](https://en.wikipedia.org/wiki/JSON_Web_Token) (JWTs) and come with a set of claims, which are a mix of standard OIDC claims and provider-specific claims
+
+IdPs must support customizing the audience claim (`aud`) and Trusted Publisher implementations must only accept OIDC tokens with a service-specific audience claim (ie “pypi”).
+
+IdPs all have their own claim sets for their OIDC tokens. Creating a Trusted Publisher for a provider requires knowledge about the provider’s internal implementation and behavior when determining which subsets constitute a sufficiently trusted set to provide to users. Trusted Publisher implementations must not rely exclusively on a generic set of claims (i.e. only `sub` + `iss` + `aud`) as these do not constitute a sufficiently trusted set for many IdPs.
+
+IdPs vary their sets of claims depending on the situation the OIDC token is used in, for example GitHub has special claims for reusable workflows or for different workflow trigger types. Providers have also been known to change their claims with varying amounts of fore-warning or documentation about the changes.
+
+Trusted Publisher implementations may utilize the `jti` claim to prevent replaying of OIDC tokens by rejecting tokens which have been previously used.
```

Comment:
NB: `jti` isn't an obligate claim under OIDC, so not all OIDC IdPs offer it. Might be worth mentioning that not all implementations _may_ be able to use it depending on the IdPs they support, as such 🙂

Reply:
Done in https://github.com/ossf/wg-securing-software-repos/pull/43/commits/2a80793eb96d7b5fbd8e2ed81f1f305cffdca12d#diff-7cdfdd87b01128fa7086e4ca06cf76139ca1065274dc7d90c44777bdc155d616R84